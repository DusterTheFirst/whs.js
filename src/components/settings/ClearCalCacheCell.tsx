/*!
 * Copyright (C) 2018-2020  Zachary Kohnen (DusterTheFirst)
 */

import dayjs from "dayjs";
import relativeTime from "dayjs/plugin/relativeTime";
import React, { useContext, useEffect, useState } from "react";
import { CalendarContext, ReloadFunctionContext } from "../../contexts";
import { RedCell } from "../../styles/components/tableview";

dayjs.extend(relativeTime);

/** A settings cell for clearing the calendar cache */
export default function ClearCalCacheCell() {
    const calendar = useContext(CalendarContext);
    const [fromNow, setFromNow] = useState(dayjs(calendar.updated).fromNow());
    const load = useContext(ReloadFunctionContext);

    useEffect(() => {
        const interval = setInterval(() => setFromNow(dayjs(calendar.updated).fromNow()), 1000);

        return () => clearInterval(interval);
    }, []);

    const clearCalendarCache = () => {
        load(true).catch((e) => console.warn("Failed to reload app", e));
    };

    return (
        <RedCell
            title="Clear Calendar Cache"
            detail={`Last update: ${fromNow}`}
            cellStyle="Subtitle"
            onPress={clearCalendarCache}
        />
    );
}