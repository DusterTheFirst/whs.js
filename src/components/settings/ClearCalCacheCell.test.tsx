/*!
 * Copyright (C) 2018-2020  Zachary Kohnen (DusterTheFirst)
 */

import React from "react";
import { render } from "react-native-testing-library";
import ClearCalCacheCell from "./ClearCalCacheCell";

// TODO: Test press

it("Renders <ClearCalCacheCell/>", () => {
    const comp = render(<ClearCalCacheCell/>);

    expect(comp.toJSON()).toMatchSnapshot();
});