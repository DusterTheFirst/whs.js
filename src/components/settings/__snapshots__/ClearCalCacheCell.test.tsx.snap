// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <ClearCalCacheCell/> 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#FFF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "minHeight": 44,
            "paddingLeft": 15,
            "paddingRight": 20,
          },
          Object {
            "backgroundColor": "#FFF",
          },
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingVertical": 10,
            },
            Object {
              "paddingVertical": 4,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Text
            allowFontScaling={true}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontSize": 16,
                  "letterSpacing": -0.32,
                },
                Object {
                  "color": "#000",
                },
                Array [
                  Object {
                    "color": "#ff5050",
                  },
                ],
              ]
            }
          >
            Clear Calendar Cache
          </Text>
          <Text
            allowFontScaling={true}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 11,
                  "letterSpacing": 0.066,
                },
                Object {
                  "color": "#000",
                },
                Object {},
              ]
            }
          >
            Last update: 50 years ago
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
